### PR TITLE
Add admin settings general docs for remote state access

### PR DESCRIPTION
**Pull request by omarismail**
This PR adds the documentation for

* Admin Settings API
* Workspaces API for remote state consumers
* Admin General Settings Descriptions

Pictures will be put in comments

### CODE REVIEW

File: content/source/docs/cloud/api/admin/settings.html.md [hunk 1]
```diff
@@ -86,6 +87,7 @@ Key path                                                                 | Type
...
 `data.attributes.send-passing-statuses-for-untriggered-speculative-plans`| bool     | `true`                    | When set to `true`, workspaces automatically send passing commit statuses for any pull requests that don't affect their tracked files.
 `data.attributes.allow-speculative-plans-on-pull-requests-from-forks`    | bool     | `false`                   | When set to `false`, [speculative plans][] are not run on pull requests from forks of a repository. This setting is available in Terraform Enterprise versions v202005-1 or later. It is currently supported for the following VCS providers: GitHub.com, GitHub.com (OAuth), GitHub Enterprise, Bitbucket Cloud, Azure DevOps Server, Azure DevOps Services. To learn more about this setting, refer to the [documentation](/docs/enterprise/admin/general.html#allow-speculative-plans-on-pull-requests-from-forks)
+`data.attributes.default-remote-state-access`    | bool     | `true`                   | When set to `true`, new workspaces will deafult to allowing remote state access to them.
```
Comment (radditude) [inline content/source/docs/cloud/api/admin/settings.html.md:90]: I'd love to link out to something here that gives a little more information about what this setting actually does! Similar to the way the description for the "allow speculative plans on pull requests from forks" setting links out to https://www.terraform.io/docs/enterprise/admin/general.html#allow-speculative-plans-on-pull-requests-from-forks.

Unfortunately, we haven't written that general description of "what is this admin setting even" yet, so there's nothing to link to 😆 I'd lean towards either adding it to this PR if you're up for it or updating this PR once it's ready.

File: content/source/docs/cloud/api/admin/settings.html.md [hunk 2]
```diff
@@ -59,7 +59,8 @@ curl \
       "api-rate-limiting-enabled": true,
       "api-rate-limit": 30,
       "send-passing-statuses-for-untriggered-speculative-plans": true,
-      "allow-speculative-plans-on-pull-requests-from-forks": false
+      "allow-speculative-plans-on-pull-requests-from-forks": false,
+      "default-remote-state-access": true
```
Comment (omarismail) [inline content/source/docs/cloud/api/admin/settings.html.md:63]: image

File: content/source/docs/cloud/api/admin/settings.html.md [hunk 3]
```diff
@@ -86,6 +87,7 @@ Key path                                                                 | Type
...
 `data.attributes.send-passing-statuses-for-untriggered-speculative-plans`| bool     | `true`                    | When set to `true`, workspaces automatically send passing commit statuses for any pull requests that don't affect their tracked files.
 `data.attributes.allow-speculative-plans-on-pull-requests-from-forks`    | bool     | `false`                   | When set to `false`, [speculative plans][] are not run on pull requests from forks of a repository. This setting is available in Terraform Enterprise versions v202005-1 or later. It is currently supported for the following VCS providers: GitHub.com, GitHub.com (OAuth), GitHub Enterprise, Bitbucket Cloud, Azure DevOps Server, Azure DevOps Services. To learn more about this setting, refer to the [documentation](/docs/enterprise/admin/general.html#allow-speculative-plans-on-pull-requests-from-forks)
+`data.attributes.default-remote-state-access`    | bool     | `true`                   | When set to `true`, new workspaces will default to allowing remote state access to them. To learn more about this setting, refer to the [documentation](/docs/enterprise/admin/general.html#default-global-remote-state).
```
Comment (omarismail) [inline content/source/docs/cloud/api/admin/settings.html.md:90]: image

File: content/source/docs/cloud/api/admin/settings.html.md [hunk 4]
```diff
@@ -126,7 +128,8 @@ curl \
       "api-rate-limiting-enabled": true,
       "api-rate-limit": 50,
       "send-passing-statuses-for-untriggered-speculative-plans": true,
-      "allow-speculative-plans-on-pull-requests-from-forks": false
+      "allow-speculative-plans-on-pull-requests-from-forks": false,
+      "default-remote-state-access": true
```
Comment (omarismail) [inline content/source/docs/cloud/api/admin/settings.html.md:132]: image

File: content/source/docs/enterprise/admin/general.html.md [hunk 5]
```diff
@@ -51,4 +51,21 @@ However, if this results in sending too many status checks to your VCS provider
...
+Workspace is being created, than this setting is ignored. When the
+`global_remote_state` parameter is **not** passed when a Workspace is being
+created, then the value of `global_remote_state` is set to the value of this
+setting.
+
+
+This setting defaults to `true`. Which implies that the `global_remote_state` of
+a Workspace, when the parameter is not passed, will default to `true`. If this
+is updated to `false`, then that would mean that `global_remote_state` of
+a Workspace, when the parameter is not passed, will default to `false`.
```
Comment (omarismail) [inline content/source/docs/enterprise/admin/general.html.md:71]: image

File: content/source/docs/cloud/api/workspaces.html.md [hunk 6]
```diff
@@ -66,6 +66,7 @@ Key path                                      | Type    | Default   | Descriptio
 `data.attributes.terraform-version`           | string  | (nothing) | The version of Terraform to use for this workspace. Upon creating a workspace, the latest version is selected unless otherwise specified (e.g. `"0.11.1"`).
 `data.attributes.trigger-prefixes`            | array   | `[]`      | List of repository-root-relative paths which should be tracked for changes, in addition to the working directory.
 `data.attributes.working-directory`           | string  | (nothing) | A relative path that Terraform will execute within. This defaults to the root of your repository and is typically set to a subdirectory matching the environment when multiple environments exist within the same repository.
+`data.attributes.global-remote-state`         | boolean | true      | Determines whether or not the remote state of this workspace can be accessed by anyone or by explicit set of remote state consumer. It defaults to `true`, meaning any other workspace can access it. If you set it to `false`, then you must explicitly add remote state consumers to this workspace. See [Remote State Consumers](/docs/cloud/api/workspaces.html#get-remote-state-consumers) for how to do that.
```
Comment (omarismail) [inline content/source/docs/cloud/api/workspaces.html.md:69]: image

File: content/source/docs/cloud/api/workspaces.html.md [hunk 7]
```diff
@@ -199,6 +200,7 @@ _Without a VCS repository_
       "trigger-prefixes": [],
       "vcs-repo": null,
       "working-directory": "",
+      "global-remote-state": true,
```
Comment (omarismail) [inline content/source/docs/cloud/api/workspaces.html.md:203]: image

File: content/source/docs/cloud/api/workspaces.html.md [hunk 8]
```diff
@@ -340,6 +343,7 @@ Key path                                      | Type           | Default
 `data.attributes.terraform-version`           | string         | (previous value) | The version of Terraform to use for this workspace.
 `data.attributes.trigger-prefixes`            | array          | (previous value) | List of repository-root-relative paths which should be tracked for changes, in addition to the working directory.
 `data.attributes.working-directory`           | string         | (previous value) | A relative path that Terraform will execute within. This defaults to the root of your repository and is typically set to a subdirectory matching the environment when multiple environments exist within the same repository.
+`data.attributes.global-remote-state`         | boolean        | true             | Determines whether or not the remote state of this workspace can be accessed by anyone or by explicit set of remote state consumer. It defaults to `true`, meaning any other workspace can access it. If you set it to `false`, then you must explicitly add remote state consumers to this workspace. See [Remote State Consumers](/docs/cloud/api/workspaces.html#get-remote-state-consumers) for how to do that.
```
Comment (omarismail) [inline content/source/docs/cloud/api/workspaces.html.md:346]: image

File: content/source/docs/cloud/api/workspaces.html.md [hunk 9]
```diff
@@ -1180,6 +1193,302 @@ $ curl \
...
+
+
+### Sample Request
+
+```shell
+$ curl \
+  --header "Authorization: Bearer $TOKEN" \
+  --header "Content-Type: application/vnd.api+json" \
+  https://app.terraform.io/api/v2/workspaces/ws-SihZTyXKfNXUWuUa/relationships/remote-state-consumers
+```
```
Comment (omarismail) [inline content/source/docs/cloud/api/workspaces.html.md:1214]: image

File: content/source/docs/cloud/api/workspaces.html.md [hunk 10]
```diff
@@ -1180,6 +1193,302 @@ $ curl \
...
+
+This PATCH endpoint requires a JSON object with the following properties as a request payload.
+
+Properties without a default value are required.
+
+| Key path                | Type   | Default | Description             |
+| --------------------    | ------ | ------- | ----------------------- |
+| `workspace_external_id` | string |         | The Workspace ID.       |
+| `data[].type`           | string |         | Must be `"workspaces"`. |
+| `data[].id`             | string |         | Must be the ID of the Workspace that is the remote state consumer. |
```
Comment (omarismail) [inline content/source/docs/cloud/api/workspaces.html.md:1349]: image

File: content/source/docs/cloud/api/workspaces.html.md [hunk 11]
```diff
@@ -1180,6 +1193,302 @@ $ curl \
...
+
+This POST endpoint requires a JSON object with the following properties as a request payload.
+
+Properties without a default value are required.
+
+| Key path                | Type   | Default | Description             |
+| --------------------    | ------ | ------- | ----------------------- |
+| `workspace_external_id` | string |         | The Workspace ID.       |
+| `data[].type`           | string |         | Must be `"workspaces"`. |
+| `data[].id`             | string |         | Must be the ID of the Workspace that is the remote state consumer. |
```
Comment (omarismail) [inline content/source/docs/cloud/api/workspaces.html.md:1403]: image

File: content/source/docs/cloud/api/workspaces.html.md [hunk 12]
```diff
@@ -1180,6 +1193,302 @@ $ curl \
...
```
Comment (omarismail) [inline content/source/docs/cloud/api/workspaces.html.md:1458]: image

**Review by nfagerlund — APPROVED** [review]
@omarismail I've added some edits on top of your branch, hopefully that's ok! With those changes included, I think this is good to go... but then, I _would,_ wouldn't I 😆

**Comment by nfagerlund** [comment]
(I also rebased, just cause it's been a minute)

**Comment by nfagerlund** [comment]
Ignore the failing link check, it's because of anchors added in a parallel PR.

**Review by radditude — APPROVED** [review]

**Comment by omarismail** [comment]
> @omarismail I've added some edits on top of your branch, hopefully that's ok! With those changes included, I think this is good to go... but then, I _would,_ wouldn't I 😆

Much better additions 😅 , thanks for adding it!

**Pull request merged**